ge/org.osate.ge/src/org/osate/ge/DiagramExporter.java

```diff
@@ -0,0 +1,296 @@
+package org.osate.ge;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
+
+import javax.imageio.ImageIO;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.e4.core.contexts.EclipseContextFactory;
+import org.eclipse.emf.common.command.AbstractCommand;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.transaction.TransactionalEditingDomain;
+import org.eclipse.graphiti.datatypes.ILocation;
+import org.eclipse.graphiti.mm.algorithms.GraphicsAlgorithm;
+import org.eclipse.graphiti.mm.algorithms.Rectangle;
+import org.eclipse.graphiti.mm.algorithms.styles.Point;
+import org.eclipse.graphiti.mm.pictograms.Connection;
+import org.eclipse.graphiti.mm.pictograms.ConnectionDecorator;
+import org.eclipse.graphiti.mm.pictograms.ContainerShape;
+import org.eclipse.graphiti.mm.pictograms.Diagram;
+import org.eclipse.graphiti.mm.pictograms.FreeFormConnection;
+import org.eclipse.graphiti.mm.pictograms.Shape;
+import org.eclipse.graphiti.platform.IDiagramBehavior;
+import org.eclipse.graphiti.services.Graphiti;
+import org.eclipse.graphiti.ui.services.GraphitiUi;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.osate.ge.internal.diagram.runtime.AgeDiagram;
+import org.osate.ge.internal.diagram.runtime.DiagramSerialization;
+import org.osate.ge.internal.diagram.runtime.layout.DiagramElementLayoutUtil;
+import org.osate.ge.internal.diagram.runtime.layout.LayoutInfoProvider;
+import org.osate.ge.internal.graphiti.AgeDiagramTypeProvider;
+import org.osate.ge.internal.graphiti.AgeFeatureProvider;
+import org.osate.ge.internal.graphiti.diagram.ColoringProvider;
+import org.osate.ge.internal.graphiti.diagram.GraphitiAgeDiagram;
+import org.osate.ge.internal.services.ExtensionRegistryService;
+import org.osate.ge.internal.ui.editor.AgeDiagramBehavior;
+import org.osate.ge.internal.util.ProjectUtil;
+import org.osgi.framework.FrameworkUtil;
+
+public class DiagramExporter {
+	/**
```

Comment:
The description is a bit long. The description should start with what the method does. Since the parameters for the methods are listed separately, they should not be explicitly described as part of the method description without a strong reason. For example:

Converts an AADL diagram file to a PNG image.

AADL should be in all upper case as well.